ext/ddtrace_profiling_native_extension/clock_id_from_pthread.c

```diff
@@ -42,4 +43,31 @@ VALUE clock_id_for(DDTRACE_UNUSED VALUE _self, VALUE thread) {
   }
 }
 
+thread_cpu_time_id thread_cpu_time_id_for(VALUE thread) {
+  rb_nativethread_id_t thread_id = pthread_id_for(thread);
+  clockid_t clock_id;
+
+  int error = pthread_getcpuclockid(thread_id, &clock_id);
+
+  if (error == 0) {
+    return (thread_cpu_time_id) {.valid = true, .clock_id = clock_id};
+  } else {
+    // TODO: Include the error code in some way in the output?
+    return (thread_cpu_time_id) {.valid = false};
+  }
+}
+
+thread_cpu_time thread_cpu_time_for(thread_cpu_time_id thread_cpu_time_id) {
```

Comment:
I'm a bit confused buy the duplicate naming in the arguments `(thread_cpu_time_id thread_cpu_time_id) {`.
I think it would be more readable if the variable name was different. One possible suggestion: `time_id`?

Reply:
How can you be confused that the function named `thread_cpu_time_...` returns a `thread_cpu_time` and takes a `thread_cpu_time_...` called `thread_cpu_time_...`? 🤣 

👍 Thanks for the suggestion! Applied in 343879de7